### PR TITLE
Release GuaCaml v0.05

### PR DESCRIPTION
**Pull request by JoanThibault**
New release of GuaCaml

- graph related modules have been largely refactore
- added export to GraphViz
- introduced some tiny meta-programming-like features

### CODE REVIEW

**Comment by JoanThibault** [comment]
sorry, but I do not understand the error in the `macos-homebrew` case

**Comment by kit-ty-kate** [comment]
the macos error can be ignored.

File: packages/Snowflake/Snowflake.0.02.02/opam [hunk 1]
```diff
@@ -6,8 +6,8 @@ license: "LGPL-3.0-only with OCaml-LGPL-linking-exception"
 homepage: "https://gitlab.com/boreal-ldd/snowflake"
 bug-reports: "https://gitlab.com/boreal-ldd/snowflake"
 depends: [
-	"ocaml" {>= "4.08"}
-	"GuaCaml" {>= "0.03"}
+	"ocaml" {>= "4.08" & < "5.0"}
```
Comment (kit-ty-kate) [inline packages/Snowflake/Snowflake.0.02.02/opam:9]: This package seems to compile just fine on 5.0. Only some previous versions of GuaCaml failed
```suggestion
	"ocaml" {>= "4.08"}
```

**Comment by JoanThibault** [comment]
I fixed previous versions of GuaCaml

File: packages/GuaCaml/GuaCaml.0.04/opam [hunk 2]
```diff
@@ -6,7 +6,7 @@ license: "LGPL-3.0-only with OCaml-LGPL-linking-exception"
 homepage: "https://gitlab.com/boreal-ldd/guacaml"
 bug-reports: "https://gitlab.com/boreal-ldd/guacaml"
 depends: [
-	"ocaml" {>= "4.13"}
+	"ocaml" {>= "4.13" & < "5.0"}
```
Comment (kit-ty-kate) [inline packages/GuaCaml/GuaCaml.0.04/opam:9]: 0.04 seems to compile file on OCaml 5.0
```suggestion
	"ocaml" {>= "4.13"}
```

**Comment by JoanThibault** [comment]
it looks good to me =)
thank you for your help

**Comment by kit-ty-kate** [comment]
Thanks

**Pull request merged**